sw/host/provisioning/ft_lib/src/lib.rs

```diff
@@ -437,12 +453,28 @@ pub fn run_ft_personalize(
     validate_certs_chain(ca_certificate.to_str().unwrap(), &certs)
 }
 
-fn parse_and_endorse_x509_cert(tbs: Vec<u8>, ca_sk: &SecretKey<NistP256>) -> Result<Vec<u8>> {
+// This internal enum provides two different certificate signing key
+// representations, a SecretKey object in case of fake key or a String object -
+// the ID of the key used by Cloud KMS.
+enum EndorsementKey {
+    FakeKey(SecretKey<NistP256>),
+    CkmsKey(String),
+}
+
+fn parse_and_endorse_x509_cert(tbs: Vec<u8>, key: &EndorsementKey) -> Result<Vec<u8>> {
```

Comment:
I am not sure if creating these KeyWrapper and EndorsementKey enums is quite a Rust ideomatic way of implementing support for two signing modes, I am open for  improvement suggestions.

Reply:
That looks fine to me.